packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -763,18 +763,136 @@ String _debugCompareFloats(String labelA, double valueA, String labelB, double v
+/// is [RenderSliverPadding], which indents its child. A sliver-to-sliver render
+/// object such as this must construct a [SliverConstraints] object for its
+/// child, then must take its child's [SliverGeometry] and use it to form its
+/// own [geometry].
+///
+/// The other common kind of one-child sliver is a sliver that has a single
+/// [RenderBox] child. An example of that would be [RenderSliverToBoxAdapter],
+/// which lays out a single box and sizes itself around the box. Such a sliver
+/// must use its [SliverConstraints] to create a [BoxConstraints] for the
+/// child, lay the child out (using the child's [layout] method), and then use
+/// the child's [RenderBox.size] to generate the sliver's [SliverGeometry].
+///
+/// The most common kind of sliver though is one with multiple children. The
+/// most straight-forward example of this is [RenderSliverList], which arranges
+/// its children one after the other in the main axis direction. As with the
+/// one-box-child sliver case, it takes its [constraints] to create a
+/// [BoxConstraints] for the children, and then it uses the aggregate
+/// information from all its children to generate its [geometry]. Unlike the
+/// one-child cases, however, it is judicious in which children it actually lays
+/// out (and later paints). If the scroll offset is 1000 pixels, and it
+/// previously determined that the first three children are each 400 pixels
+/// tall, then it will skip the first two and start the layout with its third
+/// child.
+///
+/// ### Layout
+///
+/// As they are laid out, slivers decide their size and the position of the next
+/// sliver, as well as the position of each of their children, based on input
+/// from the viewport such as the scroll offset.
+///
+/// For example, a sliver that just paints a box 100 pixels high would say it
+/// was 100 pixels tall when the scroll offset was zero, but would say it was 25
+/// pixels tall when the scroll offset was 75 pixels, and would say it was zero
+/// pixels high when the scroll offset was 100 pixels or more. (This is assuming
+/// that [SliverConstraints.remainingPaintExtent] was more than 100 pixels.)
+///
+/// The various dimensions that are provided as input to this system are in the
+/// [constraints]. They are described in detail in the documentation for the
+/// [SliverConstraints] class.
+///
+/// The [performLayout] function must take these [constraints] and create a
+/// [SliverGeometry] object that it must then assign to the [geometry] property.
+/// The different dimensions of the geometry that can be configured are
+/// described in detail in the documentation for the [SliverGeometry] class.
+///
+/// ### Painting
+///
+/// In addition to implementing layout, a sliver must also implement painting.
+/// This is achieved by overriding the [paint] method.
+///
+/// The [paint] method is called with an [Offset] to the top-left corner of the
+/// sliver, _regardless of the axis direction_.
```

Comment:
I'd say something about the [Canvas] coordinate system, otherwise it's not clear what offset is relative to.